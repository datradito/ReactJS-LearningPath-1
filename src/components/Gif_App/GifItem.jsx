import PropTypes from 'prop-types';

export default function GifItem({ gif }) {
    return (
        <>
            <div className='card'>
                <img src={gif.url} alt={gif.title}></img>
                <h4 className='capitalize'>{gif.title}</h4>
            </div>
        </>
    )
}
GifItem.propTypes = {
    gif: PropTypes.object.isRequired
}