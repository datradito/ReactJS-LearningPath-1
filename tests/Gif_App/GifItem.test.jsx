/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import GifItem from '../../src/tests/Gif_App/GifItem';

describe('Pruebas de GifItem', () => {
    const gif = {title: 'Naruto', url: 'https://www.giantbomb.com/a/uploads/scale_small/3/33873/1700999-naruto.png', id: '1'};

    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifItem gif={gif} /> );
        expect( container ).toMatchSnapshot( );
    });
    
    test('Debe mostrar el src y alt correspondiente', () => {
       render( <GifItem gif={gif} /> );
       const {src, alt} = screen.getByRole('img');
       expect( src ).toBe( gif.url )
       expect( alt ).toBe( gif.title )
    });
}) 
