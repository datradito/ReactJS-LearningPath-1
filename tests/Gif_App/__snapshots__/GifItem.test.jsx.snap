// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://www.giantbomb.com/a/uploads/scale_small/3/33873/1700999-naruto.png"
    />
    <h4
      class="capitalize"
    >
      Naruto
    </h4>
  </div>
</div>
`;
